src/pubkey.cpp

```diff
@@ -4,19 +4,182 @@
 
 #include "pubkey.h"
 
-#include "eccryptoverify.h"
+#include <secp256k1.h>
+#include <secp256k1_recovery.h>
 
-#include "ecwrapper.h"
+namespace
+{
+secp256k1_context* secp256k1_context_verify = NULL;
```

Comment:
That's indeed ugly, but it isn't supposed to happen: all but one instance of this is global (and global initializers are guaranteed to not run in parallel), and the one other is in init.cpp, before any parallel code runs. I've considered putting a lock around it, but it would make libconsensus depend on boost again (until c++11).